From 26979d6a0f70a49658d74380c4631f102cb0ebc0 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:15 +0800
Subject: [PATCH 2/9] GSI Android 17 compatibility:
 0002-Don-t-crash-on-status-UNSUPPORTED_REMOTE_OR_LMP_FEAT.patch

Change-Id: I62bb3284042e99ffc513ae0ab5913398144caa29
---
 system/gd/hci/hci_layer.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/system/gd/hci/hci_layer.cc b/system/gd/hci/hci_layer.cc
index fd85769..c4f765e 100644
--- a/system/gd/hci/hci_layer.cc
+++ b/system/gd/hci/hci_layer.cc
@@ -260,8 +260,10 @@ struct HciLayer::impl {
     using WaitingFor = CommandQueueEntry::WaitingFor;
     WaitingFor waiting_for = command_queue_.front().waiting_for_;
     CommandStatusView status_view = CommandStatusView::Create(event);
-    if (is_vendor_specific && (is_status && waiting_for == WaitingFor::COMPLETE) &&
-        (status_view.IsValid() && status_view.GetStatus() == ErrorCode::UNKNOWN_HCI_COMMAND)) {
+    if ((is_vendor_specific && (is_status && waiting_for == WaitingFor::COMPLETE) &&
+        (status_view.IsValid() && status_view.GetStatus() == ErrorCode::UNKNOWN_HCI_COMMAND)) ||
+        ((is_status && waiting_for == WaitingFor::COMPLETE) &&
+        (status_view.IsValid() && status_view.GetStatus() == ErrorCode::UNSUPPORTED_REMOTE_OR_LMP_FEATURE))) {
       // If this is a command status of a vendor specific command, and command complete is expected,
       // we can't treat this as hard failure since we have no way of probing this lack of support at
       // earlier time. Instead we let the command complete handler handle a empty Command Complete
